libs/remote_storage/src/lib.rs

```diff
@@ -331,6 +331,28 @@ pub trait RemoteStorage: Send + Sync + 'static {
         cancel: &CancellationToken,
     ) -> Result<Download, DownloadError>;
 
+    /// Same as download, but with encryption if the backend supports it (e.g. SSE-C on AWS).
+    async fn download_with_encryption(
+        &self,
+        from: &RemotePath,
+        opts: &DownloadOpts,
```

Comment:
we should put this into the `DownloadOpts` IMO

Reply:
https://github.com/neondatabase/neon/pull/11599/commits/7c2b2325f1619f96c2473d5cdbdb732e7e8aca55, I suppose we can also have something like `UploadOpts` and I can do it when I refactor the pageserver read/write path.